tool/tsh/common/tsh.go

```diff
@@ -2086,14 +2086,36 @@ func onLogout(cf *CLIConf) error {
 		proxyHost = cf.Proxy
 	}
 
-	switch {
-	// Proxy and username for key to remove.
-	case proxyHost != "" && cf.Username != "":
-		tc, err := makeClient(cf)
+	tc, err := makeClient(cf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var SAMLSingleLogoutURL string
+	clt, err := tc.ConnectToCluster(cf.Context)
+	if err != nil {
+		log.
+			Warnf("Failed to retrieve cluster client for user %s, SAML single logout will be skipped: %v", active.Username, err)
+	}
+
+	var user types.User
+	// Only run this if the ConnectToCluster above was successful.
+	if err == nil {
+		user, err = clt.AuthClient.GetUser(cf.Context, active.Username, false)
 		if err != nil {
-			return trace.Wrap(err)
+			log.
+				Warnf("Failed to retrieve user details for user %s, SAML single logout will be skipped: %v", active.Username, err)
 		}
```

Comment:
The general go convention is that on error, the normal return value must not be inspected and provides no guarantees at all, so we can't assume that `user` is nil there.
```suggestion
				Warnf("Failed to retrieve user details for user %s, SAML single logout will be skipped: %v", active.Username, err)
			user = nil
		}
```